demos/coreMQTT/mqtt_demo_connection_sharing.c

```diff
@@ -917,14 +973,87 @@ static MQTTStatus_t prvResumeSession( bool xSessionPresent )
 
 /*-----------------------------------------------------------*/
 
+static int32_t prvGenerateRandomNumber()
+{
+    const uint32_t ulMultiplier = 0x015a4e35UL, ulIncrement = 1UL;
+
+    ulNextRand = ( ulMultiplier * ulNextRand ) + ulIncrement;
+    return( ( int32_t ) ( ulNextRand >> 16UL ) & 0x7fffUL );
+}
+
+/*-----------------------------------------------------------*/
+
+static BaseType_t prvSeedRandomNumberGenerator()
```

Comment:
I see that we are having copies of the same code in a few places. Is it possible to avoid this redundancy?

Reply:
It is a valid concern, I think that the utility wrapper of generating a random number with PKCS11 can be part of the `corePKCS11` repository so that both the `amazon-freertos` and `FreeRTOS/FreeRTOS` can utilize it. 
What are your thoughts on it @lundinc2 ?

For now, that change would be beyond the scope of this PR.

Reply:
I have commonized the logic of using PKCS11 functions into a wrapper function, `xPkcs11GenerateRandomNumber`, and created a separate library named `AFR::pkcs11_helpers` that the demos link against.